core/sdk/src/extn/client/extn_client.rs

```diff
@@ -236,21 +236,36 @@ impl ExtnClient {
                 Self::handle_single(message, self.response_processors.clone());
             } else if message.payload.is_event() {
                 let is_main = self.sender.get_cap().is_main();
-                if !is_main {
-                    if let Some(context) = RippleContext::is_ripple_context(&message.payload) {
-                        trace!(
-                            "Received ripple context in {} message: {:?}",
-                            self.sender.get_cap().to_string(),
-                            message
-                        );
-                        {
-                            let mut ripple_context = self.ripple_context.write().unwrap();
-                            ripple_context.deep_copy(context);
+                if is_main // This part of code is for the main ExntClient to handle
```

Comment:
This new section of code handles the below expectation for Main's extension client to forward the event between one extension to the other.

Authorization to check whether a given Extension has the authority to send the event to the other extension is defined in the manifest with `ExtnSender forward_event` checks for the contract permissions. So this code is basically a pass through from and extension to main and then to the actual extension.

We will implement a separate change to just emit events from extension to extension without needing to go through main similar to how we handle the response for requests.